test/unit/app/storage.test.js

```diff
@@ -0,0 +1,55 @@
+const { BlobServiceClient } = require('@azure/storage-blob')
+const { getBlob, resetClient } = require('../../../app/storage')
+const { streamToBuffer } = require('../../../app/lib/streamToBuffer')
+const { storage } = require('../../../app/config')
+
+jest.mock('@azure/storage-blob')
+jest.mock('@azure/identity')
+jest.mock('../../../app/lib/streamToBuffer')
+
+const mockFromConnectionString = jest.fn(() => ({
+  getContainerClient: jest.fn(() => ({
+    getBlobClient: jest.fn(() => ({
+      download: jest.fn(() => ({
+        readableStreamBody: 'fakeStream'
+      }))
+    }))
+  }))
+}))
+BlobServiceClient.fromConnectionString = mockFromConnectionString
+
+describe('Blob Storage Service', () => {
+  beforeEach(() => {
+    streamToBuffer.mockResolvedValue('fakeBufferContent')
+  })
+
+  afterEach(() => {
+    resetClient()
+    jest.resetAllMocks()
+    jest.resetModules()
+  })
+
+  it('should initialize client with connection string and return buffer content', async () => {
+    storage.connectionString = 'fakeConnectionString'
+    storage.useConnectionString = true
+
+    const blobContent = await getBlob('fakeFile.txt')
+
+    expect(BlobServiceClient.fromConnectionString).toHaveBeenCalledWith('fakeConnectionString')
```

Comment:
This will fail but I care less bout what stub was called and more about the expected return in the line below.